### PR TITLE
Use encodeInt32 instead of encoding 32-bit integers as objects, same for 64-bit

### PR DESCRIPTION
**Pull request by pzw2041**
Use encodeInt32 instead of encoding 32-bit integers as objects, same for 64-bit integers

### CODE REVIEW

**Comment by YuanZhu-apple** [comment]
Looks good to me!

File: ResearchKit/Common/ORKHelpers.h [hunk 1]
```diff
@@ -72,8 +72,11 @@
 #define ORK_DECODE_INTEGER(d,x) _ ## x = [d decodeIntegerForKey:@STRINGIFY(x)]
 #define ORK_ENCODE_INTEGER(c,x) [c encodeInteger:_ ## x forKey:@STRINGIFY(x)]
 
-#define ORK_ENCODE_UINT32(c,x) [c encodeObject:[NSNumber numberWithUnsignedLongLong:_ ## x] forKey:@STRINGIFY(x)]
-#define ORK_DECODE_UINT32(d,x) _ ## x = (uint32_t)[(NSNumber *)[d decodeObjectForKey:@STRINGIFY(x)] unsignedLongValue]
+#define ORK_ENCODE_UINT32(c,x) [c encodeInt32:_ ## x forKey:@STRINGIFY(x)]
+#define ORK_DECODE_UINT32(d,x) _ ## x = [d decodeInt32ForKey:@STRINGIFY(x)]
+
+#define ORK_ENCODE_UINT64(c,x) [c encodeInt64:_ ## x forKey:@STRINGIFY(x)]
+#define ORK_DECODE_UINT64(d,x) _ ## x = [d decodeInt64ForKey:@STRINGIFY(x)]
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKHelpers.h]: We don't have any application for these, so let's not add them.

File: ResearchKit/Common/ORKHelpers.h [hunk 2]
```diff
@@ -72,8 +72,11 @@
 #define ORK_DECODE_INTEGER(d,x) _ ## x = [d decodeIntegerForKey:@STRINGIFY(x)]
 #define ORK_ENCODE_INTEGER(c,x) [c encodeInteger:_ ## x forKey:@STRINGIFY(x)]
 
-#define ORK_ENCODE_UINT32(c,x) [c encodeObject:[NSNumber numberWithUnsignedLongLong:_ ## x] forKey:@STRINGIFY(x)]
-#define ORK_DECODE_UINT32(d,x) _ ## x = (uint32_t)[(NSNumber *)[d decodeObjectForKey:@STRINGIFY(x)] unsignedLongValue]
+#define ORK_ENCODE_UINT32(c,x) [c encodeInt32:_ ## x forKey:@STRINGIFY(x)]
+#define ORK_DECODE_UINT32(d,x) _ ## x = [d decodeInt32ForKey:@STRINGIFY(x)]
```
Comment (jwe-apple) [inline ResearchKit/Common/ORKHelpers.h]: I'm hesitant without a test verifying that the game result that uses this works correctly for a seed of 0xffffffff. Please add a sanity check for that serialization / deserialization  using NSKeyedArchiver to the tests. It's best to add to the ResearchKit unit tests directly, even though most of the serialization tests are actually in the ORKTest project.

**Comment by jwe-apple** [comment]
Closing this. You can make a new pull request if you feel strongly.

**Pull request closed without merging**